core/src/main/scala/org/apache/spark/sql/rapids/tool/EventProcessorBase.scala

```diff
@@ -432,16 +436,7 @@ abstract class EventProcessorBase[T <: AppBase](app: T) extends SparkListener wi
       app: T,
       event: SparkListenerStageCompleted): Unit = {
     logDebug("Processing event: " + event.getClass)
-    val stage = app.getOrCreateStage(event.stageInfo)
-    stage.completionTime = event.stageInfo.completionTime
-    stage.failureReason = event.stageInfo.failureReason
-    stage.duration = ProfileUtils.optionLongMinusOptionLong(stage.completionTime,
-      stage.info.submissionTime)
-    val stageAccumulatorIds = event.stageInfo.accumulables.values.map { m => m.id }.toSeq
-    stageAccumulatorIds.foreach { accumId =>
-      val existingStages = app.accumulatorToStages.getOrElse(accumId, Set.empty)
-      app.accumulatorToStages.put(accumId, existingStages + event.stageInfo.stageId)
-    }
+    app.getOrCreateStage(event.stageInfo)
```

Comment:
Encapsulate the initialization and teh claculation of duration within the retrieval of the stage object.